extensions/stats/config.h

```diff
@@ -0,0 +1,384 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <google/protobuf/util/json_util.h>
+#include "absl/container/flat_hash_map.h"
+#include "extensions/common/context.h"
+#include "extensions/common/node_info.pb.h"
+#include "extensions/stats/config.pb.h"
+
+// WASM_PROLOG
+#ifndef NULL_PLUGIN
+#include "api/wasm/cpp/proxy_wasm_intrinsics.h"
+
+#else  // NULL_PLUGIN
+
+#include "extensions/common/wasm/null/null.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+namespace Null {
+namespace Plugin {
+#endif  // NULL_PLUGIN
+
+// END WASM_PROLOG
+
+namespace Stats {
+
+using StringView = absl::string_view;
+
+constexpr StringView Sep = "#";
+
+// The following need to be std::strings because the receiver expects a string.
+const std::string unknown = "unknown";
+const std::string vSource = "source";
+const std::string vDest = "destination";
+const std::string vMTLS = "mutual_tls";
+const std::string vNone = "none";
+const std::string vDash = "-";
+
+using google::protobuf::util::JsonParseOptions;
+using google::protobuf::util::Status;
+
+struct Node {
+  common::NodeInfo node_info;
+  // key computed from the
+  std::string key;
+
+  Node(common::NodeInfo nodeInfo) {
```

Comment:
A single param constructor usually declare as `explicit` to avoid auto converting from `common::NodeInfo`